spatialdata/_core/aggregate.py

```diff
@@ -0,0 +1,164 @@
+from __future__ import annotations
+
+import anndata as ad
+import dask.dataframe as ddf
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+from scipy import sparse
+
+import spatialdata as sd
+from spatialdata._core.models import PointsModel, ShapesModel, get_schema
+
+
+def aggregate(
+    values: ddf.DataFrame | gpd.GeoDataFrame,
+    by: gpd.GeoDataFrame,
+    id_key: str,
+    *,
+    value_key: str | None = None,
+    agg_func: str = "mean",
+) -> ad.AnnData:
```

Comment:
It would be cool to set `region`, `region_key` and `instance_key` in the returned `adata`, but `region` and `region_key` require a `sdata` in the function call, so here only `instance_key` can be set.

Reply:
So, I was thinking this could return a `SpatialData` where the region is `by`.